### DISCUSSION TITLE
Should we be less conservative and a bit bolder with Grails 7 source code styling?

### DISCUSSION THREAD

**Discussion by gsartori** [apache/grails-core#13791] [General]
Grails 7 is THE release. You know it, I know it, and 7 is a beutiful number (pair it with 21 - the current LTS JVM release - and we have the universe singing and dancing with us :)

I am aware that this topic was previously discussed in our thursday's meetings.
Yet I have the feeling this is the right time to be bolder than usual.

I have the feeling this is the moment to uniform the source code formatting. True, it could make things a bit confusing when comparing files between commits to fix things. **But I think that going forward, having a better, prettier, more readable and workable codebase is much more valuable.**

That said, instead of approaching this issue up-front, taking big decisions, I'm opening this discussion to change things on a "per case" basis.

In the comment section we can add our feelings on some specific cases so we can discuss and take small decisions diluted in time.

**Comment by jamesfredley** [comment]
The min and max Java versions are going to follow Spring and a min of Java 17 provides broader applicability.  https://start.grails.org/ allows simple 21 and 23 configuration.

https://newrelic.com/resources/report/2024-state-of-the-java-ecosystem
https://www.jetbrains.com/lp/devecosystem-2023/java/

https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-Versions#jdk-version-range
https://docs.spring.io/spring-boot/system-requirements.html

**Reply by gsartori** [reply]
I realize the first introductory line is a bit misleading, I didn’t mean to discuss the supported Java versions (!!).

The idea here is to just discuss the coding style 👍🏻

**Comment by amondel2** [comment]
@gsartori I am talking about code format in ticket #13754

**Reply by gsartori** [reply]
@amondel2 this `if` issue was the first one I noticed as soon as I downloaded the Grails sourcecode and I agree on your `keyword` vs `funciton` argument.

Could this be the first "tuning" we do on Grails coding style?

@jeffscottbrown can we add this for discussion on the next meeting?

**Reply by jeffscottbrown** [reply]
> @jeffscottbrown can we add this for discussion on the next meeting?

Of course.  Thank you for flagging it.

**Comment by dauer** [comment]
Checkstyle and CodeNarc was added and the code cleaned up in this commit https://github.com/apache/grails-core/pull/14925/

**Comment by jdaugherty** [comment]
I'm going to close this as 7 has been released and we did end up changing the styling.
